RocketChat/Rocket.Chat.ReactNative
# feat: report user

## Proposed changes

the feature to report a user was added on the server side with this PR -> https://github.com/RocketChat/Rocket.Chat/pull/29818

The user should be able to report other user from a DM or from a message in any channel

## Issue(s)	


## How to test or reproduce


- The server that you will test should be greater than or equal to 6.4.0

Test one:
- Start a DM
- Click at the header of DM
- Then go to the user profile
- you should see the button to report with red color, click it
- you will be redirected to the Report User View
- should write something
- and submit
- if the submit is success, you should be redirected to the DM

Test two:
- Inside any type of channel
- Click at the avatar or the username of other user
- You will be redirected to  the Report User View
- should write something
- and submit
- if the submit is success, you should be redirected to the channel

## Screenshots

https://github.com/RocketChat/Rocket.Chat.ReactNative/assets/47038980/f565c292-187b-48b5-9dd2-12f507dee8ae


## Types of changes




## Checklist



## Further comments

[TC-599]

[TC-599]: https://rocketchat.atlassian.net/browse/TC-599?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
